File: EPRO.Library.v3.5/EPRO.Library.v3.5/Objects/EDecimal.cs
using System;
using Microsoft.VisualBasic.CompilerServices;

namespace ELibrary.Standard.Objects
{



    /// <summary>
    /// Decimal is smaller in range but has more accuracy than double
    /// </summary>
    /// <remarks></remarks>
    public class EDecimal
    {

        /// <summary>
        /// Converts An Object to Double
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static decimal valueOf(string obj)
        {
            try
            {
                if (obj is null || (obj.Trim() ?? "") == (string.Empty ?? ""))
                    return 0m;
                return Conversions.ToDecimal(obj);
            }
            catch (Exception ex)
            {
                return 0m;
            }
        }


        /// <summary>
        /// Converts objects to double
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static decimal valueOf(object obj)
        {
            if (obj is null)
                return 0m;
            double d = (double)Conversions.ToDecimal(obj);
            if (obj is string)
                return valueOf(Conversions.ToString(obj));
            if (obj is DBNull)
                return 0m;
            if (obj is double && Conversions.ToDouble(obj) <= (double)decimal.MaxValue)
                return Conversions.ToDecimal(obj);
            if (obj is long)
                return Conversions.ToLong(obj);
            if (obj is int)
                return Conversions.ToInteger(obj);
            if (obj is short)
                return Conversions.ToInteger(obj);
            if (obj is bool)
                return EShort.valueOf(Conversions.ToBoolean(obj));
            if (obj is decimal)
                return Conversions.ToDecimal(obj);
            return 0m;    // REM Cant convert this
        }
    }
}

File: ELibrary.Standard.VB/Objects/EDecimal.vb
Namespace Objects



    ''' <summary>
    ''' Decimal is smaller in range but has more accuracy than double
    ''' </summary>
    ''' <remarks></remarks>
    Public Class EDecimal

        ''' <summary>
        ''' Converts An Object to Double
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As String) As Decimal
            Try
                If obj Is Nothing OrElse obj.Trim() = String.Empty Then Return 0
                Return CDec(obj)

            Catch ex As Exception
                Return 0

            End Try
        End Function


        ''' <summary>
        ''' Converts objects to double
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Object) As Decimal

            If obj Is Nothing Then Return 0
            Dim d As Double = CDec(obj)
            If TypeOf obj Is String Then Return valueOf(CType(obj, String))
            If TypeOf obj Is DBNull Then Return 0D
            If TypeOf obj Is Double AndAlso CDbl(obj) <= Decimal.MaxValue Then Return CDec(obj)
            If TypeOf obj Is Long Then Return (CType(obj, Long))
            If TypeOf obj Is Int32 Then Return CInt(obj)
            If TypeOf obj Is Int16 Then Return CInt(obj)
            If TypeOf obj Is Boolean Then Return EShort.valueOf(CType(obj, Boolean))
            If TypeOf obj Is Decimal Then Return CDec(obj)

            Return 0    REM Cant convert this

        End Function



    End Class

End Namespace